src/test/java/org/opensearch/security/dlic/rest/api/UserApiTest.java

```diff
@@ -75,6 +75,28 @@ public void testSecurityRoles() throws Exception {
         checkGeneralAccess(HttpStatus.SC_OK, "newuser", "newuser");
     }
 
+    @Test
+    public void testParallelPutRequests() throws Exception {
+        
+        setup();
+        
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+
+        HttpResponse[] responses = rh.executeMultipleAsyncPutRequest(10, ENDPOINT + "/internalusers/test1", "{\"password\":\"test1\"}");
+        boolean created = false;
+        for (HttpResponse response : responses) {
+            int sc = response.getStatusCode();
+            if (sc == HttpStatus.SC_CREATED) {
```

Comment:
why test is covering only Created or Conflict scenario? code might still throw internal error based on line 298?

Reply:
In the test case we are receiving parallel put requests. One of the request should return 400 and the rest of them should return 409. The original code in `onFailure` catch all the exceptions including 409 and return them 500. So after the fix, the test should expect 409 and 400 in the normal situation.

Reply:
I believe we are expecting 200, 409 and 500 HTTP status code as response. But for test we are only testing for 200 and 409. Does that mean we will never see 500?

Reply:
oops sorry, for 400 i mean 200

Reply:
I don't think 500 will happen in this situation.